trie/hexary.py

```diff
@@ -248,6 +230,30 @@ def root_node(self, value):
     #
     # Utils
     #
+
+    @contextlib.contextmanager
+    def _prune_node(self, node):
+        """
+        Prune the given node if context exits cleanly.
+        """
+        if self.is_pruning:
+            # node is mutable, so capture the key for later pruning now
+            prune_key, node_body = self._node_to_db_mapping(node)
+        else:
+            node_body = None
+
+        # Nothing to prune
+        if node_body is None:
+            yield
+        else:
+            # Prune only if no exception is raised
+            try:
+                yield
+            except:
```

Comment:
I don't think you need this try/except block at all (and we probably end up taking a performance hit for it).

```python
try:
    thing()
except:
    raise
else:
    something()

# equivalent to
thing()
something()
```

Reply:
#86